polyfills/fm/client/fmradio.js

```diff
@@ -0,0 +1,553 @@
+        operation: 'get',
+        name: 'antennaAvailable'
+      };
+
+      var request = {};
+      request.serialize = serializeRequest.bind(request, data);
+
+      navConnPromise.then(conn => conn.sendObject(request));
+      return this.fakeAntennaAvailable;
+    },
+
+    set: function(newValue) {
+      antennaAvailable = newValue;
+    }
+  });
+
+  Object.defineProperty(window.navigator.mozFMRadio, 'frequency', {
+    get: function() {
+      // Current frequency value should be updated
+      if (typeof frequency !== 'undefined' && frequency !== null) {
+        return frequency;
+      }
+
+      var data = {
+        operation: 'get',
+        name: 'frequency'
+      };
+
+      var request = {};
+      request.serialize = serializeRequest.bind(request, data);
+
+      navConnPromise.then(conn => conn.sendObject(request));
+      // Let's assume that we're in the lower bound frequency
+      return this.fakeFrequencyLowerBound;
+    },
+
+    set: function(newValue) {
+      frequency = newValue;
+    }
+  });
+
+  Object.defineProperty(window.navigator.mozFMRadio, 'frequencyUpperBound', {
+    get: function() {
+      if (typeof frequencyUpperBound !== 'undefined' &&
+        frequencyUpperBound !== null) {
+          return frequencyUpperBound;
+      }
+
+      var data = {
+        operation: 'get',
+        name: 'frequencyUpperBound'
+      };
+
+      var request = {};
+      request.serialize = serializeRequest.bind(request, data);
+
+      navConnPromise.then(conn => conn.sendObject(request));
+      return this.fakeFrequencyUpperBound;
+    },
+
+    set: function(newValue) {
+      frequencyUpperBound = newValue;
+    }
+  });
+
+  Object.defineProperty(window.navigator.mozFMRadio, 'frequencyLowerBound', {
+    get: function() {
+      if (typeof frequencyLowerBound !== 'undefined' &&
+        frequencyLowerBound !== null) {
+          return frequencyLowerBound;
+      }
+
+      var data = {
+        operation: 'get',
+        name: 'frequencyLowerBound'
+      };
+
+      var request = {};
+      request.serialize = serializeRequest.bind(request, data);
+
+      navConnPromise.then(conn => conn.sendObject(request));
+      return this.fakeFrequencyLowerBound;
+    },
+
+    set: function(newValue) {
+      frequencyLowerBound = newValue;
+    }
+  });
+
+  Object.defineProperty(window.navigator.mozFMRadio, 'channelWidth', {
+    get: function() {
+      if (typeof channelWidth !== 'undefined' && channelWidth !== null) {
+        return channelWidth;
+      }
+
+      var data = {
+        operation: 'get',
+        name: 'channelWidth'
+      };
+
+      var request = {};
+      request.serialize = serializeRequest.bind(request, data);
+
+      navConnPromise.then(conn => conn.sendObject(request));
+      return this.fakeChannelWidth;
+    },
+
+    set: function(newValue) {
+      channelWidth = newValue;
+    }
+  });
+
+  Object.defineProperty(window.navigator.mozFMRadio, 'onfrequencychange', {
```

Comment:
Besides the same comment of the previous setters, this also can be refactored to not c&p all the common parts... something like:

_onchange: function(changeType, cb) {
  this['_on' + changeType] = cb;
  this['_on' + changeType + 'Id'] = 
....
}
set onfrequencychange(cb) {
  this._onchange('frequencychange', cb);
}